backend/src/models/reports/index.js

```diff
@@ -620,6 +620,108 @@ const getLessonsLearnedReport = (fiscalYear, projectId, portfolioIds) => {
   return results;
 };
 
+/**
+ * Gets the data for a Contract Summary Report
+ *
+ * @param   {number} contractId Contract id to limit report to.
+ * @returns {any}
+ */
+const getContractSummaryReport = (contractId) => {
+  return knex("data.contract as contract")
+    .select({
+      contract: "contract.*",
+      start_date: knex.raw(`TO_CHAR(contract.start_date :: DATE, '${dateFormat}')`),
+      end_date: knex.raw(`TO_CHAR(contract.end_date :: DATE, '${dateFormat}')`),
+      total_contract: knex.raw("contract.total_fee_amount + contract.total_expense_amount"),
+      supplier_name: "supplier_name",
+      internal_coding: "internal_coding.*",
+      portfolio: "portfolio.*",
+      portfolio_name: "portfolio.portfolio_name",
+    })
+    .leftJoin("data.supplier as supplier", { "supplier.id": "contract.supplier_id" })
+    .leftJoin("data.sid_internal_coding as internal_coding", {
+      "contract.id": "internal_coding.contract_id",
+    })
+    .leftJoin("data.portfolio as portfolio", { "portfolio.id": "internal_coding.portfolio_id" })
+    .where("contract.id", contractId);
+};
+
+/**
+ * Gets the contract invoices for a specific contract by id
+ *
+ * @param   {number} contractId Contract id to limit report to.
+ * @returns {any[]}
+ */
+const getContractInvoices = (contractId) => {
+  const results = knex("data.invoice as invoice")
+    .select({
+      fiscal: knex.min("fiscal_year"),
+      billing_period: knex.min("billing_period"),
+      invoice_date: knex.raw(`MIN(TO_CHAR(invoice_date:: DATE, '${dateFormat}'))`),
+      invoice_number: "invoice_number",
+      invoice_amount: knex.raw("SUM(unit_amount * rate)"),
+    })
+    .leftJoin("data.contract as contract", { "contract.id": "invoice.contract_id" })
+    .leftJoin("data.fiscal_year as fiscal_year", { "fiscal_year.id": "invoice.fiscal" })
+    .leftJoin("data.invoice_detail as invoice_detail", {
+      "invoice.id": "invoice_detail.invoice_id",
+    })
+    .where("invoice.contract_id", contractId)
+    .groupBy("invoice_number")
+    .orderBy("invoice_number");
+  return results;
+};
+
+/**
+ * Gets the contract payment summary for a specific contract by id and fiscal year
+ *
+ * @param   {number}     contractId Contract id to limit report to.
+ * @param   {string}     fiscalYear Fiscal year to summarize over.
+ * @returns {any[]}
+ */
+const getContractPaymentSummary = (contractId, fiscalYear) => {
+  return knex("data.invoice_detail as detail")
+    .select({
+      fiscal: "fiscal_year",
+      fees_invoiced: knex.raw("SUM(unit_amount * rate)"),
+      rees_remaining: knex.raw("MIN(total_fee_amount) - SUM(unit_amount * rate)"),
```

Comment:
is this supposed to be fees_remaining... not rees_remaining
```suggestion
      fees_remaining: knex.raw("MIN(total_fee_amount) - SUM(unit_amount * rate)"),
```